src/shared/request.ts

```diff
@@ -0,0 +1,75 @@
+import { ZodType } from 'zod';
+import { Protocol } from './protocol.js';
+import { Request, Notification, Result, Task, GetTaskResult } from '../types.js';
+import { isTerminal } from './task.js';
+
+const DEFAULT_POLLING_INTERNAL = 5000;
```

Comment:
Was this meant to be `DEFAULT_POLLING_INTERVAL`?

And speaking of internal, shouldn't this be configurable? It also might make sense to allow people to override the polling interval on the client-side and disregard what the server suggests since respecting it is only a SHOULD in the SEP, but at the very least you should be able to configure the default when the server doesn't specify anything.